go/exocom/src/client_registry/client_registry.go

```diff
@@ -0,0 +1,77 @@
+package clientRegistry
+
+import "encoding/json"
+
+// Client is the combination of a client name, service type and internal namespace
+type Client struct {
+	ClientName        string
+	ServiceType       string
+	InternalNamespace string
+}
+
+// Route is an entry in the routes
+type Route struct {
+	Receives          []string
+	Sends             []string
+	InternalNamespace string
+}
+
+// ClientRegistry manages which clients are connected to exocom,
+// and what messages each client can send and receive
+type ClientRegistry struct {
+	Routing       map[string]Route
+	Clients       map[string]Client
+	subscriptions *SubscriptionManager
```

Comment:
This one probably also goes into the broadcast subsystem. Maybe add convenience methods to add and look up entries to it as well?